tuned_lens/plotting/prediction_trajectory.py

```diff
@@ -538,6 +542,42 @@ def cross_entropy(self, **kwargs) -> TrajectoryStatistic:
             stats=stats,
         )
 
+    def rank(self, show_ranks=False, **kwargs) -> TrajectoryStatistic:
+        """The rank of the targets among the predictions.
+
+        Args:
+            show_ranks: Whether to show the the rank of the target or the top token.
+            **kwargs: are passed to largest_prob_labels.
+
+        Returns:
+            A TrajectoryStatistic with the rank of the targets among the predictions.
+        """
+        if self.targets is None:
+            raise ValueError("Cannot compute rank without targets.")
+
+        # Yes I know this is not the most efficient way to do this
```

Comment:
It's also kind of confusing, so why are you doing it this way?

Why not `np.sum(self.log_probs > _select_values_along_seq_axis(self.log_probs, self.targets)[..., None, :], axis=-1) + 1`?

Is it to make sure the rank of each token would be different?

Reply:
Of course you can just say "nah it's just how I thought of doing it" and leav it like that.

Reply:
I'm a bit embarrassed about how long it took me to implement this correctly. I'm sure you could implement it better but I don't think performance is super important here.

Reply:
If the code you gave passes the correctness test I'm happy to use it

Reply:
Added PR #106